src/main/java/frc/robot/RobotContainer.java

```diff
@@ -4,48 +4,81 @@
 
 package frc.robot;
 
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.auto.NamedCommands;
 import edu.wpi.first.math.MathUtil;
 import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.RunCommand;
 import edu.wpi.first.wpilibj2.command.button.Trigger;
 import frc.robot.commands.BasicDriveCommand;
 import frc.robot.constants.RobotConstants.DriveConstants.OIConstants;
 import frc.robot.subsystems.drive.Drivetrain;
+import frc.robot.subsystems.vision.Vision;
 import frc.utils.Vector;
 
 public class RobotContainer {
   private Drivetrain m_robotDrive;
+  private Vision m_robotVision;
 
   // The driver's controller
-  XboxController m_driverController;
+  private XboxController m_driverController;
+  private SendableChooser<Command> autoChooser;
+
+  private Vector leftInputVec;
+  private Vector rightInputVec;
 
   public RobotContainer() {
+    m_robotVision = new Vision();
     m_robotDrive = new Drivetrain();
     m_driverController = new XboxController(OIConstants.kDriverControllerPort);
+    autoChooser = AutoBuilder.buildAutoChooser();
+    leftInputVec = new Vector();
+    rightInputVec = new Vector();
 
     configureBindings();
+    registerCommands();
 
     m_robotDrive.setDefaultCommand(
         // The left stick controls translation of the robot.
         // Turning is controlled by the X axis of the right stick.
         new RunCommand(
-            () ->
-                m_robotDrive.drive(
-                    new Vector(
-                        MathUtil.applyDeadband(
-                            -m_driverController.getLeftY(), OIConstants.kDriveDeadband),
-                        MathUtil.applyDeadband(
-                            -m_driverController.getLeftX(), OIConstants.kDriveDeadband)),
-                    new Vector(
-                        MathUtil.applyDeadband(
-                            -m_driverController.getRightX(), OIConstants.kDriveDeadband),
-                        MathUtil.applyDeadband(
-                            -m_driverController.getRightY(), OIConstants.kDriveDeadband)),
-                    m_driverController.getRightBumper(),
-                    m_driverController.getAButton()),
+            () -> {
+              updateInput();
+              m_robotDrive.drive(
+                  leftInputVec,
```

Comment:
Why are you passing two empty vectors here?

Reply:
declaring fewer vectors saves memory, but I'll add a comment to explain it